src/query_plan/operation.rs

```diff
@@ -1869,12 +2758,19 @@ impl TryFrom<&NormalizedInlineFragmentSelection> for InlineFragment {
     }
 }
 
-impl From<&NormalizedFragmentSpreadSelection> for FragmentSpread {
-    fn from(val: &NormalizedFragmentSpreadSelection) -> Self {
-        Self {
-            fragment_name: val.data().fragment_name.to_owned(),
-            directives: val.data().directives.deref().to_owned(),
-        }
+impl TryFrom<&NormalizedFragmentSpreadSelection> for FragmentSpread {
```

Comment:
It looks like this impl is infallible. Why change `From` → `TryFrom`?

Reply:
indeed, this was a leftover from copy+paste of the inline fragment trait impl (which can fail due to the selection set conversion)

Reply:
[6f3d768](https://github.com/apollographql/federation-next/pull/239/commits/6f3d76812aa9c52e2b1b4b92a5570df6c6ba2995)